ofrak_core/ofrak/gui/server.py

```diff
@@ -260,6 +264,68 @@ async def run_until_cancelled(self):  # pragma: no cover
         finally:
             await self.runner.cleanup()
 
+    @exceptions_to_http(SerializedError)
+    async def init_chunked_root_resource(self, request: Request) -> Response:
+        name = request.query.get("name")
+        size_param = request.query.get("size")
+        size = int(size_param) if size_param is not None else None
+        if name is None:
+            raise HTTPBadRequest(reason="Missing resource name from request")
+        if size is None:
+            raise HTTPBadRequest(reason="Missing chunk size from request")
+        root_resource: Resource = await self._ofrak_context.create_root_resource(
+            name, bytearray(b"\x00" * size), (File,)
+        )
+        self.resource_builder[root_resource.get_id().hex()] = (
+            root_resource,
+            memoryview(bytearray(b"\x00" * size)),
+        )
+        return json_response(root_resource.get_id().hex())
+
+    @exceptions_to_http(SerializedError)
+    async def root_resource_chunk(self, request: Request) -> Response:
+        id = request.query.get("id")
+        start_param = request.query.get("start")
+        start = int(start_param) if start_param is not None else None
+        end_param = request.query.get("end")
+        end = int(end_param) if end_param is not None else None
+        if id is None:
+            raise HTTPBadRequest(reason="Missing resource id from request")
+        if start is None:
+            raise HTTPBadRequest(reason="Missing chunk start from request")
+        if end is None:
+            raise HTTPBadRequest(reason="Missing chunk end from request")
+        chunk_data = await request.read()
+        _, data = self.resource_builder[id]
+        data[start:end] = chunk_data
+        return json_response([])
+
+    @exceptions_to_http(SerializedError)
+    async def create_chunked_root_resource(self, request: Request) -> Response:
+        id = request.query.get("id")
+        name = request.query.get("name")
+        if id is None:
+            return HTTPBadRequest(reason="Missing root resource `id` from request")
+        if name is None:
+            return HTTPBadRequest(reason="Missing root resource `name` from request")
+
+        try:
+            root_resource, data = self.resource_builder[id]
+            script_str = rf"""
+            if root_resource is None:
+                root_resource = await ofrak_context.create_root_resource_from_file("{name}")"""
+            root_resource.queue_patch(Range(0, len(bytearray(data))), bytearray(data))
```

Comment:
Is it necessary to cast `data` to a `bytearray` to get the length? I would worry about it doing expensive constructor operations just to throw away the object after it gets the length. If it is necessary to cast for getting the lenght, maybe store the casted version in a variable instead of casting twice?

```suggestion
            root_resource.queue_patch(Range(0, len(data)), bytearray(data))
```